### ISSUE TITLE
SourceCodeDisclosureFileInclusion slow with large responses

### ISSUE THREAD

**Issue by NVolcz** [zaproxy/zaproxy#2545]
Sometimes during active scans zap freezes almost completely. 

I ran a very quick samling with VisualVM and from the first look it seems like the culprit in this case was 
SourceCodeDisclosureFileInclusion when using HirshbergMatcher.getLCS() with large responses.

**Comment by thc202** [comment]
Correct, that's a known performance issue (that's the reason why LDAPInjection scanner was demoted to alpha status, zaproxy/zap-extensions#117).

**Comment by NVolcz** [comment]
What would be a good solution for this performance issue? Find a faster algorithm or maybe use some lookup table or exclude known bad responses before threshold comparing?

**Comment by kingthorin** [comment]
If someone can point to a faster algorithm for finding Longest Common Substring we'd be open to evaluating that.

Another option would be to ignore (skip) responses over a certain size (perhaps based on Threshold setting).

How big are the responses where you're noticing the issue, and what content type are they?

<sub>Code ref: https://github.com/zaproxy/zap-extensions/blob/22bb0c2334f92d7d535834331dcdd7864657f6b4/src/org/zaproxy/zap/extension/ascanrulesAlpha/SourceCodeDisclosureFileInclusion.java</sub>

**Comment by NVolcz** [comment]
Based on some guesswork:
HTML pages: 30KB
Some larger JS: 100-500KB
A couple of binary files: -20MB

Could we skip checking files that exceeds the "Large response" threshold?
